### ISSUE TITLE
Reimplement in C++

### ISSUE THREAD

**Issue by serin-delaunay** [serin-delaunay/hueperlin#1]
The Python implementation of the hueperlin algorithm and demo is rather slow, and the Python SFML wrapper used was not as portable as expected.

The main motivation for writing this in Python was the difficulty in finding Free C/C++ implementations of the Jacobi theta function. However, https://github.com/fredrik-johansson/arb has a Jacobi theta implementation in C, and recently switched from GPL to LGPL, which makes it a viable alternative to https://github.com/fredrik-johansson/mpmath, the original provider.

HUSL has been renamed to HSLuv, and there is a GLSL implementation at https://github.com/williammalo/hsluv-glsl which could easily be translated to C.

There are plenty of C/C++ implementations of Perlin noise; https://github.com/qknight/libnoise is particularly suitable.

Unwrapped C++ SFML is as portable as it ever was.

In summary, it is now feasible to make a fast, portable, MIT-licensed version of this algorithm and demo.
